pkg/apis/apps/v1alpha1/servicebindingrequest_types.go

```diff
@@ -63,11 +63,11 @@ type BackingServiceSelector struct {
 // +k8s:openapi-gen=true
 type ApplicationSelector struct {
 	// +optional
-	MatchLabels map[string]string `json:"matchLabels,omitempty"`
-	Group       string            `json:"group,omitempty"`
-	Version     string            `json:"version"`
-	Resource    string            `json:"resource"`
-	ResourceRef string            `json:"resourceRef"`
+	LabelSelector *metav1.LabelSelector `json:",inline"`
```

Comment:
Why an extra comma in the beginning `,inline`?

Reply:
The tag name for the field is irrelevant here. For example, if the tag is like this: `json:"labelSelector,inline"` the word `labelSelector` will not appear anywhere as the values are embedded.

There is an explanation about `omitempty` option which is somewhat similar to `inline` option.
https://golang.org/pkg/encoding/json/#Marshal

I can't find any explanation about `inline` anywhere. But to understand the behavior you can run this example: https://gist.github.com/baijum/e761bc8fb28ba3855e5d1cd118e9192b

Let me know if any other clarification required.